src/node_crypto.cc

```diff
@@ -4827,6 +4828,448 @@ void Scrypt(const FunctionCallbackInfo<Value>& args) {
 #endif  // OPENSSL_NO_SCRYPT
 
 
+class KeyPairGenerationConfig {
+ public:
+  virtual EVP_PKEY_CTX* Setup() = 0;
+  virtual bool Configure(const EVPKeyCtxPointer& ctx) {
+    return true;
+  }
+};
+
+class RSAKeyPairGenerationConfig : public KeyPairGenerationConfig {
+ public:
+  RSAKeyPairGenerationConfig(unsigned int modulus_bits, unsigned int exponent)
+    : modulus_bits_(modulus_bits), exponent_(exponent) {}
+
+  EVP_PKEY_CTX* Setup() {
+    return EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, nullptr);
+  }
+
+  bool Configure(const EVPKeyCtxPointer &ctx) {
```

Comment:
style nits: Can you use left-leaning references, and use `override` on overridden virtual methods?